### DISCUSSION TITLE
Async SQL (Relational) Databases - how does it work?

### DISCUSSION THREAD

**Discussion by JacobBITLABS** [fastapi/fastapi#4176] [Questions] [answered]
Hi! 
I have implemented the concepts from the Async SQL tutorial and have a few questions. Maybe they are stupid, so sorry about that. 
Does this implementation utilize the uvloop that FastAPI is using under the hood (as I understand), and hence optimising the organisation of the different (now async) calls to a synchronous DB connection using the loop?  There is linked to the encode/databases repository that is providing support to asyncpg driver, but as I can tell from the implementation-details in the tutorial, it is a synchronous connection? I get quite a performance boost in requests handled per second, so something is kicking in, but I suspect it to be the logic and not an optimised db connection.
I know that the asynchronous database connection is not the holy grail as pointed out by the SQLAlchemy author (https://techspot.zzzeek.org/2015/02/15/asynchronous-python-and-databases/) and hence probably not going to implement the SQLAlchemy asynchronous functions.

Thanks in advance!

**Comment by vargasjona** [comment, marked as answer]
Hello @JacobBITLABS here you can find an async configuration of database using SQLModel ORM and FastAPI https://github.com/jonra1993/fastapi-alembic-sqlmodel-async

**Reply by JacobBITLABS** [reply]
Hi, thank you very much.
I actually ended up with something that seems similar to what you have implemented. I get the best of both worlds as the sequential calls in FastAPI is executed in a uvloop. The database-connection is synchronous do to the arguments for CRUD-based applications by the SQLAlchemy author: https://techspot.zzzeek.org/2015/02/15/asynchronous-python-and-databases/ , that is backed with benchmarked tests.
